### PR TITLE
don't set bash options, use pipes instead

### PR DESCRIPTION
**Pull request by marten-seemann**
cc @mvdan

### CODE REVIEW

**Review by mvdan — APPROVED** [review]
out of curiosity, can you not just use $? directly in the if check?

**Comment by marten-seemann** [comment]
I tried that earlier and it didn't work. I tried again and now it works :(
Will update the PR.

File: action.yml [hunk 1]
```diff
@@ -15,12 +15,10 @@ runs:
         for dir in $(find . \( -name vendor -o -name '[._].*' -o -name node_modules \) -prune -o -name go.mod -print | sed 's:/go.mod$::'); do
           echo "#=#=#=# $dir #=#=#=#"
           pushd $dir > /dev/null
-          set +e -x
+          s=0
           (
```
Comment (Stebalien) [inline action.yml:18]: note: you can also say:

```
if ! (
${{ inputs.run }}
); then
status = ...
```

I think?

Comment (marten-seemann) [reply action.yml:18]: Isn't it great that there are 20 different ways to do the easiest things in bash?

Comment (marten-seemann) [reply action.yml:18]: That works, but we'd lose the exit status.

Comment (Stebalien) [reply action.yml:18]: Ah, I see. Yeah, that makes sense.

>  Isn't it great that there are 20 different ways to do the easiest things in bash?

Hm. That's _way_ too few. We should switch everything to Perl.

**Pull request merged**